common/jmeter/oidc/OIDC_Password_Grant_RBAC.jmx

```diff
@@ -0,0 +1,363 @@
+            <stringProp name="Argument.name">deployment</stringProp>
+            <stringProp name="Argument.value">${__P(deployment,&quot;active-passive&quot;)}</stringProp>
+            <stringProp name="Argument.metadata">=</stringProp>
+          </elementProp>
+        </collectionProp>
+      </Arguments>
+      <hashTree/>
+      <ThreadGroup guiclass="ThreadGroupGui" testclass="ThreadGroup" testname="Password Grant">
+        <boolProp name="ThreadGroup.delayedStart">true</boolProp>
+        <stringProp name="ThreadGroup.num_threads">${noOfThreads}</stringProp>
+        <stringProp name="ThreadGroup.ramp_time">${rampUpPeriod}</stringProp>
+        <stringProp name="ThreadGroup.duration">${timeToRun}</stringProp>
+        <longProp name="ThreadGroup.delay">0</longProp>
+        <boolProp name="ThreadGroup.same_user_on_next_iteration">true</boolProp>
+        <boolProp name="ThreadGroup.scheduler">true</boolProp>
+        <stringProp name="ThreadGroup.on_sample_error">continue</stringProp>
+        <elementProp name="ThreadGroup.main_controller" elementType="LoopController" guiclass="LoopControlPanel" testclass="LoopController" testname="Loop Controller">
+          <intProp name="LoopController.loops">-1</intProp>
+          <boolProp name="LoopController.continue_forever">false</boolProp>
+        </elementProp>
+      </ThreadGroup>
+      <hashTree>
+        <RandomVariableConfig guiclass="TestBeanGUI" testclass="RandomVariableConfig" testname="Random SP Counter">
+          <stringProp name="maximumValue">${noOfSPs}</stringProp>
+          <stringProp name="minimumValue">1</stringProp>
+          <stringProp name="outputFormat"></stringProp>
+          <boolProp name="perThread">true</boolProp>
+          <stringProp name="randomSeed"></stringProp>
+          <stringProp name="variableName">spCountIdRandom</stringProp>
+        </RandomVariableConfig>
+        <hashTree/>
+        <BeanShellPreProcessor guiclass="TestBeanGUI" testclass="BeanShellPreProcessor" testname="BeanShell PreProcessor">
+          <boolProp name="resetInterpreter">false</boolProp>
+          <stringProp name="parameters"></stringProp>
+          <stringProp name="filename"></stringProp>
+          <stringProp name="script">import java.util.Random;
+
+try {
+    int spCountId = Integer.parseInt(vars.get(&quot;spCountIdRandom&quot;));
+    int userCount = Integer.parseInt(vars.get(&quot;noOfUsersPerSP&quot;));
+    int noOfSP = Integer.parseInt(vars.get(&quot;noOfSPs&quot;));
+    
+    int minUserCount;
+    int maxUserCount;
+
+    if (spCountId == 1) {
+        minUserCount = 1;
+    } else if (spCountId == 0){
+	   log.info(&quot;No application selected&quot;);
+	   minUserCount = 1;
+    } else {
+        minUserCount = (userCount * (spCountId - 1)) + 1;
+    }
+    maxUserCount = userCount * spCountId;
+
+    Random rand = new Random();
+    int randomUserCount = rand.nextInt((maxUserCount - minUserCount) + 1) + minUserCount;
```

Comment:
why do we use this, when we have JMeter random counter option there?
check `RandomVariableConfig`